pkg/manifest/jsonmanifest/jsonmanifest_test.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package jsonmanifest_test
+
+import (
+	"net/http"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/manifest"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/swarm/test"
+)
+
+var testCases = []testCase{
+	{
+		name:    "empty-manifest",
+		entries: nil,
+	},
+	{
+		name: "one-entry",
+		entries: []e{
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-1",
+				headers:   http.Header{},
+				path:      "",
+			},
+		},
+	},
+	{
+		name: "two-entries",
+		entries: []e{
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-1.txt",
+				headers:   http.Header{"Content-Type": {"text/plain; charset=utf-8"}},
+				path:      "",
+			},
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-2.png",
+				headers:   http.Header{"Content-Type": {"image/png"}},
+				path:      "",
+			},
+		},
+	},
+	{
+		name: "nested-entries",
+		entries: []e{
+			{
+				reference: test.RandomAddress(),
+				name:      "robots.txt",
+				headers:   http.Header{"Content-Type": {"text/plain; charset=utf-8"}},
+				path:      "text",
+			},
+			{
+				reference: test.RandomAddress(),
+				name:      "1.png",
+				headers:   http.Header{"Content-Type": {"image/png"}},
+				path:      "img",
+			},
+			{
+				reference: test.RandomAddress(),
+				name:      "2.jpg",
+				headers:   http.Header{"Content-Type": {"image/jpg"}},
+				path:      "img",
+			},
+			{
+				reference: test.RandomAddress(),
+				name:      "readme.md",
+				headers:   http.Header{"Content-Type": {"text/markdown; charset=UTF-8"}},
+				path:      "",
+			},
+		},
+	},
+}
+
+// TestEntries verifies that manifests behave as expected when adding and removing entries.
+// It also tests the Length and Entry functions.
+func TestEntries(t *testing.T) {
+	// get non-trivial test case
+	tc := testCases[len(testCases)-1] // last test case
+	if len(tc.entries) < 2 {
```

Comment:
wtf... why do we need `L87-L90`?

Reply:
sorry :sweat_smile: i was paranoid that an eventual change to the `testCases` var would negatively impact `TestEntries`, since it always takes the last test case of the array. 

and i wondered: _what if this _new_ last test case was not big enough for this test to make sense any longer?_

...on second thought, if that were to be the case, i would rather this test failed.

removed